### PR TITLE
Resharding

### PR DESCRIPTION
**Pull request by alainjobart**
@enisoc @aaijazi 
Fixing replication after reparents.

### CODE REVIEW

**Comment by enisoc** [comment]
LGTM

File: go/vt/mysqlctl/replication.go [hunk 1]
```diff
@@ -69,8 +74,8 @@ func changeMasterArgs(params *sqldb.ConnParams, masterHost string, masterPort in
 }
 
 // parseSlaveStatus parses the common fields of SHOW SLAVE STATUS.
-func parseSlaveStatus(fields map[string]string) *proto.ReplicationStatus {
-	status := &proto.ReplicationStatus{
+func parseSlaveStatus(fields map[string]string) proto.ReplicationStatus {
```
Comment (aaijazi) [inline go/vt/mysqlctl/replication.go]: Why the change from pass-by-reference everywhere?

File: go/vt/tabletmanager/agent_rpc_actions.go [hunk 2]
```diff
@@ -331,28 +331,31 @@ func (agent *ActionAgent) MasterPosition(ctx context.Context) (myproto.Replicati
 	return agent.MysqlDaemon.MasterPosition()
 }
 
-// StopSlave will stop the replication
+// StopSlave will stop the replication Works both when Vitess manages
```
Comment (aaijazi) [inline go/vt/tabletmanager/agent_rpc_actions.go]: This commends needs rewording...

File: go/vt/mysqlctl/replication.go (hunk 1 above)
Comment (alainjobart) [inline go/vt/mysqlctl/replication.go]: To be consistent with proto.ReplicationPosition everywhere. It's a small structure anyway, that has one interface and 5 simple-type members.

Comment (aaijazi) [inline go/vt/mysqlctl/replication.go]: Personally, I'd favor making it consistent by doing pass-by-reference everywhere. Although it's a small structure, it may grow in the future, and we've already been bit by something like this once, so it makes me cautious...

Not a blocker though.

**Comment by aaijazi** [comment]
LGTM

File: go/vt/mysqlctl/replication.go (hunk 1 above)
Comment (alainjobart) [inline go/vt/mysqlctl/replication.go]: Then you meet with Anthony and you guys figure it out, he's the one who made me change proto.ReplicationPosition from pointer to value, as it only contains a single interface object (which is already a pointer, basically)...

Come to think of it, I don't mind one being a value and the other one a pointer. I can do that too. 

@enisoc @aaijazi what do you say?
